src/main/scala/firrtl/EmissionOption.scala

```diff
@@ -9,6 +9,18 @@ package firrtl
   */
 trait EmissionOption
 
+/** Emission customization options for memories */
+trait MemoryEmissionOption extends EmissionOption {
+  def initValue: MemoryInitValue = MemoryRandomInit()
+}
+
+sealed trait MemoryInitValue
+case class MemoryRandomInit() extends MemoryInitValue
```

Comment:
This should be a case object since it's a nullary product type.
```suggestion
case object MemoryRandomInit() extends MemoryInitValue
```

Reply:
See my other comment